### DISCUSSION TITLE
Contributing to x-drip

### DISCUSSION THREAD

**Discussion by animalmutch** [NightscoutFoundation/xDrip#2013] [General]
Hi. Firstly, big thanks for all you do with x-drip. My other half is type 1 diabetic, and has found the app truly life changing. I'm a software engineer. Most of my experience is in Python and web dev stack, but I do a bit of Java/Android development in my spare time. I have zero experience in the open source world, but have been looking for a project to get involved with and I can't think of a better project than x-drip. So, just wondering how to go about it? Just choose an issue from the backlog and submit a PR? Thanks :)

**Comment by Navid200** [comment]
Wait for @tolot27 to respond.

But, may I ask which CGM you are familiar with?  That could make a big difference.  What I mean is that if your loved one is using Libre, it would not be best if you focus on an issue that is specific to Dexcom and vice versa.  That's of course only my opinion.  Well, everything I post is my opinion.

There are issues that are common of course.

If I were you, I would first choose an issue that I was interested in.  Then, I would post here in the discussions, how you want to address it.  I don't mean necessarily technically.  What I mean is from the user perspective.
For example, you say there is no low volume profile.  I want to add a low volume profile.
This is an example I am making for a purpose.  We don't have that for a reason.  But, we unfortunately don't have all the reasoning documented.  So, we will respond and explain why we don't have it.

But, if there is no reason like that, you may hear someone saying that's wonderful.  
Still, I wouldn't spend too much time on something and open a PR.  There are times you do everything right and still your proposal is not merged.

So, I suggest as the first project, to pick something as small as possible.  Just to see how things work.

Thanks for your interest.
We need people like you

**Reply by animalmutch** [reply]
Thanks @Navid200 , seems like sound advice. My OH just got a Libre 2, after using a Libre 1 and Miaomiao for a couple of years, so have just gone through the process of setting that up. It struck me how difficult a lot of people might find the process, so doing something to help with that seems like a good start. Not being a Java aficionado, I'd also like to start with something simple. I've not looked into the code too much, but I suspect #1672 fits the bill

**Reply by Navid200** [reply]
@tzachi-dar is a developer who has the most experience with Libre.  He is also one developer who has been actively working on the xDrip code longer than any other developer.  So, there is a lot in the code everywhere that he knows about because he has created it.
If you want to work on something specific to Libre, I strongly advise that you work in close cooperation with him.
